article-analyzer/src/main/java/co/inventorsoft/academy/spring/resolver/NotificationServiceResolver.java

```diff
@@ -0,0 +1,68 @@
+package co.inventorsoft.academy.spring.resolver;
+
+import co.inventorsoft.academy.spring.model.NotificationType;
+import co.inventorsoft.academy.spring.repository.UserRepository;
+import co.inventorsoft.academy.spring.service.EmailNotificationService;
+import co.inventorsoft.academy.spring.service.NotificationService;
+import co.inventorsoft.academy.spring.service.SlackNotificationService;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.FileNotFoundException;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Data
+@AllArgsConstructor
+@Component
+public class NotificationServiceResolver {
+
+    /*private EmailNotificationService emailNF;
+    private SlackNotificationService slackNF;
+    public NotificationService getNotificationService(NotificationType type){
+        switch (type) {
+            case EMAIL -> {
+                return emailNF;
+            }
+            case SLACK -> {
+                return slackNF;
+            }
+            default -> {
+                return null;
+            }
+        }
+    }*/
+
+    private List<Class<? extends NotificationService>> notificationServices;
+
+    @Autowired
```

Comment:
avoid field injection, move it to contructor

Reply:
changed